spec/services/case_extend_sar_deadline_service_spec.rb

```diff
@@ -0,0 +1,172 @@
+        ).call
+      }
+
+      it { is_expected.to eq :ok }
+      it 'creates new SAR extension transition' do
+        expect(sar_case.state_machine)
+          .to have_received(:extend_sar_deadline!)
+          .with(
+            acting_user: manager,
+            acting_team: team_dacu,
+            final_deadline: initial_deadline + 3.days,
+            original_final_deadline: initial_deadline,
+            message: "test\nDeadline extended by 3 days"
+          )
+      end
+
+      it 'sets new SAR deadline date' do
+        expect(sar_case.external_deadline).to eq initial_deadline + 3.days
+      end
+    end
+
+    context 'after initial deadline' do
+      it 'allows retrospective extension' do
+        deadline = sar_case.initial_deadline
+
+        Timecop.travel(deadline + 100.days) do
+          result = new_service(manager, sar_case, max_extension).call
+
+          expect(result).to eq :ok
+          expect(sar_case.external_deadline).to eq deadline + max_extension.days
+          expect(sar_case.external_deadline).to be < DateTime.now
+        end
+      end
+    end
+
+    context 'validate' do
+      context 'extension period' do
+        context 'is missing' do
+          subject! { new_service(manager, sar_case, nil).call }
+
+          it { is_expected.to eq :validation_error }
+          it {
+            expect(sar_case.errors[:extension_period])
+              .to eq ["can't be blank"]
+          }
+        end
+
+        context 'is past statutory SAR extension limit' do
+          let!(:extension_period) { max_extension + 5 }
+          subject! {
+            new_service(
+              manager,
+              sar_case,
+              extension_period
+            ).call
+          }
+
+          it { is_expected.to eq :validation_error }
+          it {
+            expect(sar_case.errors[:extension_period])
+              .to eq ["can't be more than 60 days beyond the initial deadline"]
+          }
+        end
+
+        context 'is before the final deadline' do
+          subject! { new_service(manager, sar_case, -1).call }
+
+          it { is_expected.to eq :validation_error }
+          it {
+            expect(sar_case.errors[:extension_period])
+              .to eq ["can't be before the final deadline"]
+          }
+        end
+
+        context 'is within limit' do
+          subject! { new_service(manager, sar_case, max_extension).call }
+
+          it { is_expected.to eq :ok }
+        end
+      end
+
+      context 'reason' do
+        subject! { new_service(manager, sar_case, 1, reason: '').call }
+
+        it { is_expected.to eq :validation_error }
+        it {
+          expect(sar_case.errors[:reason_for_extending])
+            .to eq ["can't be blank"]
+        }
+      end
+
+      context 'exception' do
+        let(:service) { new_service(manager, sar_case, 3) }
+
+        it 'rolls-back changes' do
+          allow(sar_case).to receive(:update!).and_throw(RuntimeError)
```

Comment:
Not sure what's causing the error here. Is it something we could have more meaningful output of to help us track it down if it breaks in production?

Reply:
This was taken from tests for `CaseExtendForPITService` which tries to active the failure of the ActiveRecord transaction.

Reply:
ok, could we improve it?

Reply:
Renamed with more descriptive spec name